In [4]:
import pandas as pd


path = '~/ICNLSP_paper/data/gsap_new/'
new_path = '~/ICNLSP_paper/data/real_data/'

# File paths
file1 = path+'test.csv'
file2 = path+'validation.csv'
file3 = path+'train.csv'

# Read and concatenate
df = pd.concat([pd.read_csv(file1,delimiter='$'), pd.read_csv(file2,delimiter='$'), pd.read_csv(file3,delimiter='$')], ignore_index=True)

len(df)

94820

In [5]:
df = df[df['section_txt'].notna()]
len(df)
#test_df = test_df.drop_duplicates(subset=['section_txt','label'])

94820

In [6]:
df = df.drop_duplicates(subset=['section_txt','label'])
len(df)

6873

In [7]:
def get_label_percentage(df, cur_label):
    sample_count = df[df['label']==cur_label]
    label_percentage = sample_count*100/len(df)
    return sample_count, label_percentage

In [8]:
pos_samples = df[df['label']==1]
print('number of positive samples: '+str(len(pos_samples)))
print(f'percentage of positive samples: {len(pos_samples)*100/6873}')

number of positive samples: 1347
percentage of positive samples: 19.598428633784373


In [9]:
neg_samples = df[df['label']==0]
num_neg_samples = len(neg_samples)
print(f'number of negative samples: {num_neg_samples}')
print(f'percentage of positive samples: {num_neg_samples*100/6873}')

number of negative samples: 5526
percentage of positive samples: 80.40157136621562


The data set contains a total of $6873$ paragraphs. Among them $1347$ are positvie while the remaining $5,526$ are negative. This amounts to $19.6$\% of positive smaples and 
$80.4$\% of negative samples. 
    
Similar to what we have done with the We divide the data into train, dev and test splits haveing $80$\%, $10$\%, and $10\%$ respectively. 

We divide the 

In [10]:
import pandas as pd
from sklearn.model_selection import train_test_split

# Assuming `df` is the DataFrame from the previous step
# Split the DataFrame into 80% training and 20% testing
train_df, test_df = train_test_split(df, test_size=0.2, random_state=42)

#Analyze train and dev splits
post_count(get_label_percentage(train_df),1)
print(post_count)


ModuleNotFoundError: No module named 'sklearn'

In [11]:
import pandas as pd
import numpy as np

# Assuming `df` is your DataFrame
# Shuffle the DataFrame
df = df.sample(frac=1, random_state=42).reset_index(drop=True)

# Calculate the split index
split_index = int(len(df) * 0.8)

# Split the DataFrame
train_df = df.iloc[:split_index]
test_df = df.iloc[split_index:]

In [12]:
len(train_df)
pos_para = train_df[train_df['label']==1]
pos_percentage = len(pos_para) * 100 / len(train_df)
print(len(pos_para))
print(pos_percentage)
print(len(train_df))

1080
19.6435067297199
5498


In [14]:
len(test_df)
pos_para = test_df[test_df['label']==1]
pos_percentage = len(pos_para) * 100 / len(test_df)
print(len(pos_para))
print(pos_percentage)
print(len(test_df))
test_df.head(2)

267
19.418181818181818
1375


id  \
5498   00013_2104_08727.txt|text_unit_8   
5499  10033_2205_01863.txt|text_unit_13   

                                            section_txt  label dataset  
5498  (Q 1 ) benefit similarly from pre-training? (Q...      0     NaN  
5499  Next, we briefly introduce the relevant backgr...      0     NaN

In [44]:
# Save the splits into CSV files
train_df.to_csv(new_path+'train.csv', sep='$', index=False)  # Save training data
test_df.to_csv(new_path+'test.csv', sep='$', index=False)    # Save testing data

print("Data split and saved successfully.")

Data split and saved successfully.


In [32]:
df['dataset'].nunique()

448

In [33]:
train_df['dataset'].nunique()

390

In [34]:
test_df['dataset'].nunique()

154

In [37]:
shared_values = set(train_df['dataset']).intersection(set(test_df['dataset']))
len(shared_values)

97

In [41]:
sff = df.drop_duplicates(subset=['dataset'])

In [42]:
len(sff)

449

In [43]:
sff

id  \
0      10006_2102_07346.txt|text_unit_63   
3       00030_2107_07653.txt|text_unit_0   
4      10002_2004_03742.txt|text_unit_35   
13    00046_2105_09680.txt|text_unit_342   
22      00004_1908_04913.txt|text_unit_4   
...                                  ...   
6777  00043_1910_10683.txt|text_unit_108   
6786   00002_1810_04805.txt|text_unit_18   
6787   00032_1908_10084.txt|text_unit_39   
6792  00046_2105_09680.txt|text_unit_235   
6812    00033_2012_14740.txt|text_unit_2   

                                            section_txt  label  \
0     These previous studies have significantly adva...      0   
3     Recent progress in language model pre-training...      1   
4     We also find that increasing the constants c a...      1   
13    We frame KLUE-DP as a sequence tagging problem...      1   
22    To mitigate the race bias in the existing face...      1   
...                                                 ...    ...   
6777  Like the unsupervised objective, the pre-train...      1   
6786  There has also been work showing effective tra...      1   
6787  • CR: Sentiment prediction of customer product...      1   
6792  The Penn Treebank [84] is a constituency parse...      1   
6812  The recent progress of VrDU lies primarily in ...      1   

                        dataset  
0                           NaN  
3     weakly-supervised WikiSQL  
4                        THUCTC  
13                      KLUE-DP  
22             YFCC-100M Flickr  
...                         ...  
6777     pre-training data sets  
6786                  Ima-geNet  
6787                         CR  
6792              Penn Treebank  
6812                       VrDU  

[449 rows x 4 columns]